pkg/resources/management.cattle.io/v3/cluster/validator.go

```diff
@@ -89,6 +90,11 @@ func (a *admitter) Admit(request *admission.Request) (*admissionv1.AdmissionResp
 		return nil, fmt.Errorf("failed get old and new clusters from request: %w", err)
 	}
 
+	if request.Operation == admissionv1.Delete && oldCluster.Name == localCluster {
+		// deleting "local" cluster could corrupt the cluster Rancher is deployed in
+		return admission.ResponseBadRequest("cannot delete the local cluster"), nil
```

Comment:
non-blocking nit: "cannot delete local cluster" isn't maybe as informative as it could be. Maybe also include why in the error since I think this could be displayed to the user. Or maybe change "cannot" to "refusing".

Error message should be consistent either way.